src/framebounds.c

```diff
@@ -21,6 +21,26 @@ struct plum_rectangle * get_frame_boundaries (struct context * context, bool anc
 
 void adjust_frame_boundaries (const struct plum_image * image, struct plum_rectangle * restrict boundaries) {
   uint64_t empty_color = get_empty_color(image);
+  #define checkframe(type) do                                                                                                                             \
+    for (uint_fast32_t frame = 0; frame < image -> frames; frame ++) {                                                                                    \
+      for (size_t remaining = (size_t) boundaries[frame].top * image -> width; remaining; remaining --)                                                   \
+        if (notempty(type)) goto done ## type;                                                                                                            \
+      if (boundaries[frame].left || boundaries[frame].width != image -> width)                                                                            \
+        for (uint_fast32_t row = 0; row < boundaries[frame].height; row ++) {                                                                             \
+          for (uint_fast32_t col = 0; col < boundaries[frame].left; col ++) if (notempty(type)) goto done ## type;                                        \
+          index += boundaries[frame].width;                                                                                                               \
+          for (uint_fast32_t col = boundaries[frame].left + boundaries[frame].width; col < image -> width; col ++)                                        \
+            if (notempty(type)) goto done ## type;                                                                                                        \
+        }                                                                                                                                                 \
+      else                                                                                                                                                \
+        index += (size_t) boundaries[frame].height * image -> width;                                                                                      \
+      for (size_t remaining = (size_t) (image -> height - boundaries[frame].top - boundaries[frame].height) * image -> width; remaining; remaining --)    \
+        if (notempty(type)) goto done ## type;                                                                                                            \
+      continue;                                                                                                                                           \
```

Comment:
This `continue` is an alternative to a manually-checked `bool adjust` flag.

This control flow:

```c
for (...) {
  bool adjust = true;
  if (various stuff) goto done;
  adjust = false;
  done:
  if (adjust) finish up;
}
```

was refactored to this:

```c
for (...) {
  if (various stuff) goto done;
  continue;
  done:
  finish up;
}
```